spark/src/main/java/org/apache/iceberg/spark/data/SparkOrcWriter.java

```diff
@@ -319,8 +319,8 @@ public void addValue(int rowId, int column, SpecializedGetters data,
         // make sure the child is big enough
         cv.child.ensureSize(cv.childCount, true);
         // Add each element
-        for (int e = 0; e < cv.lengths[rowId]; ++e) {
```

Comment:
This doesn't make sense to me. Should `cv.lengths` be an integer array instead of longs? It looks like they should always be integers since the values are assigned from `ArrayData#numElements()`. That's very likely an int.

Reply:
You're right @rdblue, I've reverted the change. The underlying is `long[]`, but it is filled with only int's.